### DISCUSSION TITLE
How to use NSItemProvider with KFImage?

### DISCUSSION THREAD

**Discussion by axiel7** [onevcat/Kingfisher#1962] [Q&A] [answered]
I'm trying to implement a simple drag and drop feature.
So far I managed to drag and drop the image url link, but I want to copy the actual image.
```swift
KFImage(imageUrl)
    .onDrag {
        NSItemProvider(contentsOf: imageUrl) ?? NSItemProvider()
    }
```
There is the `NSItemProvider(object: NSItemProviderWriting)` initialiser but I have no idea how to use it with Kingfisher.

Any help is appreciated 😄

**Comment by onevcat** [comment, marked as answer]
There is an image object in the `onSuccess` callback, maybe you can try to store that and create an `NSItemProvider`. Some sample snippets:

```swift
var image: KFCrossPlatformImage?

struct ContentView : View {
  KFImage(url)
    .onSuccess { r in
      image = r.image
    }
    .onDrag({
      if let i = image {
        return .init(object: i)
      } else {
        return .init(contentsOf: url) ?? .init()
      }
    })
}
```

**Reply by axiel7** [reply]
Thank you! I didn't know about the image object
